src/transformStateWithClones.js

```diff
@@ -7,7 +7,63 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateCloneArray = [];
+  let indexLastActions = 0;
+
+  actions.forEach((action) => {
+    const { type, extraData, keysToRemove } = action;
+
+    switch (type) {
+      case 'clear':
+        stateCloneArray.push(Object.assign({}));
+        break;
+
+      case 'addProperties':
+
+        if (!stateCloneArray.length) {
```

Comment:
No need to treat the first action different from others. You should always clone the last state and apply the action.